pkg/transport/missing_san_roundtripper.go

```diff
@@ -0,0 +1,43 @@
+package transport
+
+import (
+	"net/http"
+
+	"github.com/openshift/library-go/pkg/crypto"
+)
+
+type roundTripperFunc func(req *http.Request) (*http.Response, error)
+
+func (rt roundTripperFunc) RoundTrip(r *http.Request) (*http.Response, error) {
+	return rt(r)
+}
+
+type counter interface {
+	Inc()
+}
```

Comment:
how is the counter going to be used? Shouldn't it be public since `NewMissingSANRoundTripper()` is public and accepts it as its argument?

Reply:
We don't want to expose the interface as public (it's not an API). We want consumers to dependency-inject a counter implementation that is compatible with this signature. This is the case for both native prometheus counters and k8s counters.

Example: https://github.com/openshift/oauth-server/pull/95/commits/12d2d1d34cdd4230c6be1458aa13d2e1f54e29c0#diff-26eb62ac396048ba3dc28ae5764caf68710f82c7956d15af24dfef8542c7fbc8R755